packages/adblocker/src/resources.ts

```diff
@@ -51,18 +51,12 @@ export type ResourcesDistribution = {
 
 // TODO - support empty resource body
 
-const SCRIPTLET_BODY_PREFIX = `if (typeof scriptletGlobals === 'undefined') {
+export function wrapScriptletBody(body: string, fnName: string) {
+  return `if (typeof scriptletGlobals === 'undefined') {
   var scriptletGlobals = {};
 }
-`;
-const SCRIPTLET_BODY_SUFFIX = `(...['{{1}}','{{2}}','{{3}}','{{4}}','{{5}}','{{6}}','{{7}}','{{8}}','{{9}}','{{10}}'].filter((a,i) => a !== '{{'+(i+1)+'}}').map((a) => decodeURIComponent(a)));`;
-
-export function wrapScriptletBody(
-  script: string,
-  prefix: string = SCRIPTLET_BODY_PREFIX,
-  suffix: string = SCRIPTLET_BODY_SUFFIX,
-) {
-  return prefix + script + suffix;
+${body}
+${fnName}(...['{{1}}','{{2}}','{{3}}','{{4}}','{{5}}','{{6}}','{{7}}','{{8}}','{{9}}','{{10}}'].filter((a,i) => a !== '{{'+(i+1)+'}}').map((a) => decodeURIComponent(a)));`;
```

Comment:
now I have a second though on the fnName. Alternatively we could use this syntax:
```js
(${body})(...[args])
```